tools/clang/test/HLSLFileCheck/hlsl/resource_binding/resource_in_cbuffer.hlsl

```diff
@@ -1,24 +1,61 @@
 // RUN: %dxc -E main -T ps_6_0 %s  | FileCheck %s
-// CHECK:tx0.s                             sampler      NA          NA      S0             s0     1
-// CHECK:tx1.s                             sampler      NA          NA      S1             s1     1
-// CHECK:tx0.t                             texture     f32          2d      T0             t0     1
-// CHECK:tx1.t                             texture     f32          2d      T1             t1     1
+
+// Make sure structures with only resources have resource fields removed,
+// are considered empty, take no space in CBuffer, and do not force alignment.
+// CHECK: cbuffer $Globals
+// CHECK:   struct hostlayout.$Globals
+// CHECK:       float h;                                      ; Offset:    0
+// CHECK:       struct hostlayout.struct.LegacyTex
+// CHECK:           /* empty struct */
+// CHECK:       } tx1;                                        ; Offset:    4
+// CHECK:       float i;                                      ; Offset:    4
+// CHECK:   } $Globals;                                       ; Offset:    0 Size:     8
+// CHECK: cbuffer CB0
+// CHECK:   struct hostlayout.CB0
+// CHECK:       float f;                                      ; Offset:    0
+// CHECK:       struct hostlayout.struct.LegacyTex
+// CHECK:           /* empty struct */
+// CHECK:       } tx0;                                        ; Offset:    4
+// CHECK:       float g;                                      ; Offset:    4
+// CHECK:   } CB0;                                            ; Offset:    0 Size:     8
+
+// CHECK: $Globals                          cbuffer      NA          NA     CB0            cb0     1
+// CHECK: CB0                               cbuffer      NA          NA     CB1            cb1     1
+
+// TODO: Preserve delcaration order when allocating resources from structs
+// CHECK-DAG: tx0.s                             sampler      NA          NA      S{{.}}             s{{.}}     1
+// CHECK-DAG: tx1.s                             sampler      NA          NA      S{{.}}             s{{.}}     1
+// CHECK-DAG: b0                                texture     f32         buf      T{{.}}             t{{.}}     1
+// CHECK-DAG: b1                                texture     f32         buf      T{{.}}             t{{.}}     1
+// CHECK-DAG: tx0.t                             texture     f32          2d      T{{.}}             t{{.}}     1
+// CHECK-DAG: tx1.t                             texture     f32          2d      T{{.}}             t{{.}}     1
```

Comment:
Note that this change doesn't regress resource allocation ordering, it was simply incorrect for resources in structs.  It might work out to the expected order, but might not, depending on the order of HL subscript users and such.  Stable from run to run, but not predictable.  This is a separate issue that needs fixing.  This test basically makes sure all are used, but the bindings end up arranged in an order you wouldn't expect based on the order of declaration, and since this table will be sorted by binding, CHECK-DAG is required as well.